crates/rooch-rpc-server/src/service/aggregate_service.rs

```diff
@@ -0,0 +1,167 @@
+        }
+
+        let coin_store_handle = coin_module
+            .coin_store_handle(account_addr)?
+            .expect("Get coin store handle should succ");
+
+        let mut result = vec![];
+        if let Some(coin_type) = coin_type {
+            let coin_store_type = format!(
+                "{}::coin::CoinStore<0x{}>",
+                ROOCH_FRAMEWORK_ADDRESS_LITERAL,
+                coin_type.to_canonical_string()
+            );
+            let key = resource_tag_to_key(&StructTag::from_str(coin_store_type.as_str())?);
+            let keys = vec![key];
+            let mut states = self
+                .rpc_service
+                .get_annotated_states(AccessPath::table(coin_store_handle, keys))
+                .await?;
+
+            let state = states.pop().flatten().expect("State expected return value");
+            let annotated_coin_store =
+                AnnotatedCoinStore::new_from_annotated_move_value(state.move_value)?;
+
+            let balance_info =
+                BalanceInfo::new_with_default(coin_type, annotated_coin_store.get_coin_value());
+            result.push(Some((None, balance_info)))
+        } else {
+            //TODO If the coin store list exceeds MAX_RESULT_LIMIT_USIZE, consider supporting traverse or pagination
+            let states = self
+                .rpc_service
+                .list_annotated_states(
+                    AccessPath::table_without_keys(coin_store_handle),
+                    cursor,
+                    limit,
+                )
+                .await?;
+
+            let mut data = states
+                .into_iter()
+                .map(|item| {
+                    item.map(|(key, state)| {
+                        let coin_store =
+                            AnnotatedCoinStore::new_from_annotated_move_value(state.move_value)
+                                .expect("AnnotatedCoinStore expected return value");
+
+                        let coin_type =
+                            get_first_ty_as_struct_tag(coin_store.get_coin_struct_type())
+                                .expect("Coin type expected get_first_ty_as_struct_tag succ");
+                        let balance_info =
+                            BalanceInfo::new_with_default(coin_type, coin_store.get_coin_value());
+                        (Some(key), balance_info)
+                    })
+                })
+                .collect::<Vec<_>>();
+
+            result.append(&mut data);
+        };
+
+        for (_key, balance_info) in result.iter_mut().flatten() {
+            let coin_info = coin_info_table
+                .get(&balance_info.coin_type)
+                .expect("Get coin info by coin type expected return value");
+            balance_info.symbol = coin_info.symbol.clone();
+            balance_info.decimals = coin_info.decimals;
+        }
+
+        Ok(result)
+    }
+}
+
+lazy_static! {
+    static ref RUNTIME: Runtime = tokio::runtime::Builder::new_multi_thread()
```

Comment:
The tokio runtime needs to pass from the main function to the service. This runtime is another runtime, not in the same runtime which start the service.

Reply:
It will encounter hung up if only one runtime, when use `futures::executor::block_on`.

I tried use tokio::block_up, but also failed when `call_function` execute.

Reply:
The main function runtime should be built with multi-thread. If this approach works, we can refactor it later.